File: src/main/java/com/runhwani/runmate/service/GroupService.java
// src/main/java/com/runhwani/runmate/service/GroupService.java
package com.runhwani.runmate.service;

import com.runhwani.runmate.dto.request.group.GroupRequest;
import com.runhwani.runmate.model.Group;

public interface GroupService {
    /**
     * 새 그룹을 생성하고 생성된 도메인 객체를 반환
     */
    Group createGroup(GroupRequest request);
}

File: RunMate/data/src/main/java/com/D107/runmate/data/remote/api/GroupService.kt
package com.D107.runmate.data.remote.api

import GroupResponse
import com.D107.runmate.data.remote.common.ApiResponse
import com.D107.runmate.data.remote.common.ServerResponse
import com.D107.runmate.data.remote.request.group.GroupCreateRequest
import com.D107.runmate.data.remote.response.group.GroupJoinResponse
import com.squareup.moshi.Json
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST


interface GroupService {
    @POST("groups/create")
    suspend fun createGroup(
        @Body createGroupRequest: GroupCreateRequest
    ): Response<ServerResponse<GroupResponse>>

    @GET("groups/current")
    suspend fun getCurrentGroup(): Response<ServerResponse<GroupResponse>>

    @DELETE("groups/leave")
    suspend fun leaveGroup(): Response<ServerResponse<Any>>

    @POST("groups/join")
    suspend fun joinGroup(
        @Body
        @Json (name= "inviteCode")
        inviteCode: String
    ): Response<ServerResponse<GroupJoinResponse>>

    @POST("groups/start")
    suspend fun groupStart(): Response<ServerResponse<Any>>

    @POST("groups/finish")
    suspend fun groupFinish(): Response<ServerResponse<Any>>

    @GET("groups/hasGroupHistory")
    suspend fun hasGroupHistory(): Response<ServerResponse<Boolean>>

}